packages/zoe/src/contracts/loan/updateDebt.js

```diff
@@ -30,48 +31,72 @@ export const makeDebtCalculator = debtCalculatorConfig => {
     calcInterestFn = calculateInterest,
     originalDebt,
     loanMath,
-    periodAsyncIterable,
+    periodNotifier,
     interestRate,
+    interestPeriod,
     zcf,
     configMinusGetDebt,
   } = debtCalculatorConfig;
   let debt = originalDebt;
 
+  let lastCalculationUpdate;
+
   const {
     updater: debtNotifierUpdater,
     notifier: debtNotifier,
   } = makeNotifierKit();
 
   const getDebt = () => debt;
 
-  const config = { ...configMinusGetDebt, getDebt };
+  const config = { ...configMinusGetDebt, getDebt, interestPeriod };
 
-  const updateDebt = _state => {
-    const interest = loanMath.make(calcInterestFn(debt.value, interestRate));
-    debt = loanMath.add(debt, interest);
-    debtNotifierUpdater.updateState(debt);
-    scheduleLiquidation(zcf, config);
+  const updateDebt = state => {
```

Comment:
```suggestion
  const updateDebt = newTimestamp => {
```
`updateDebt` doesn't need the `updateCount`, so let's just pass in the timestamp

Reply:
done